ui/objects/semantic-ui/js/objects/detail/Carousel.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from "react";
+import Slider from "react-slick";
+import {
+  Modal,
+  Image,
+  Button,
+  Grid,
+  Header
+} from "semantic-ui-react";
+
+
+
+export const ImgCarousel = ({ imagesCollection, fileName}) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [selectedImageIndex, setSelectedImageIndex] = useState(0);
+  const [slidesToShow, setSlidesToShow] = useState(5);
+  
+
+  const settings = {
+    dots: true,
+    infinite: false,
+    speed: 100,
+    slidesToShow: slidesToShow,
+    slidesToScroll: 1,
+    swipeToSlide: true,
+  };
+
+  // next image
+  const handleNextImage = () => {
+    setSelectedImageIndex(
+      (prevIndex) => (prevIndex + 1) % imagesCollection.length
+    );
+  };
+
+  // previous image
+  const handlePrevImage = () => {
+    setSelectedImageIndex(
+      (prevIndex) =>
+        (prevIndex - 1 + imagesCollection.length) % imagesCollection.length
+    );
+  };
+
+  // Update slidesToShow
+  useEffect(() => {
+    const updateSlidesToShow = () => {
+      if (window.innerWidth <= 992 && window.innerWidth >= 530) {
+        setSlidesToShow(3);
+      } else if (window.innerWidth <= 530) {
+        setSlidesToShow(2);
+      } else {
+        setSlidesToShow(5);
+      }
+    };
+
+    updateSlidesToShow();
+    window.addEventListener("resize", updateSlidesToShow);
+
+    return () => {
+      window.removeEventListener("resize", updateSlidesToShow);
+    };
+  }, []);
+
+
+
+  return (
+    <>
+      <Slider {...settings}>
+        {imagesCollection?.map((image, index) => {
+          return (
+          <Image
+            key={index}
+            src={image.links.content}
+            alt={fileName(image)}
```

Comment:
I just used a function(could be found in detail/index.js) for both files and images for checking if there is a default file/image name (not set while uploading). If yes, it would use file/image's key to avoid repeating 'default_file_name' caption. This function is also used in carousel-> alt, so I could avoid the same part of code in both components. Should it be just renamed or deleted and used separately in both components?

Reply:
If so, then i would rename that utility function into something like `getCaption(image)`? :)